### PR TITLE
allow multiline mode for input with type text (close #458)

### LINKED ISSUE #458: Allow multiline mode for input with type text
Currently if we choose `text` type, hasura still allows only 1 line input.
We should allow multiline text area for multiline input also.

Even better suggestion: Allow WYSIWYG and Markdown mode as options for users to switch. And we could allow for even more formats in future (like LaText)

This is really cool feature to have, because it'll allow rich text inputing right from the Hasura console UI without any need for external system.

### PR DESCRIPTION
**Pull request by Anupam-dagar**
This adds a toggle react component which can be used to change text input to textarea and vice versa.

close #458 

### Description

What component does this PR affect? 

- [ ] Server
- [x] Console
- [ ] CLI
- [ ] Docs
- [ ] Community Content
- [ ] Build System

Requires changes from other components? If yes, please mark the components:

- [ ] Server
- [ ] Console
- [ ] CLI
- [ ] Docs
- [ ] Community Content
- [ ] Build System

### Related Issue

https://github.com/hasura/graphql-engine/issues/458

### Solution and Design

toggle_component

### Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs update
- [ ] Community content

### Checklist:

- [x] I have read the **[contributing guide](https://github.com/hasura/graphql-engine/blob/master/CONTRIBUTING.md)** and my code conforms to the guidelines.
- [ ] This change requires a change in the documentation. 
- [ ] I have updated the documentation accordingly.
- [ ] I have added required tests.

### CODE REVIEW

**Comment by Anupam-dagar** [comment]
@praveenweb I'll fix the tests.
Also for the font awesome icons, maybe we can use the following icons.
https://fontawesome.com/icons/text-width?style=solid
https://fontawesome.com/icons/text-height?style=solid
https://fontawesome.com/icons/align-justify?style=solid

**Comment by praveenweb** [comment]
@dsandip - can you suggest the final icon / position UI for the toggle functionality?

**Comment by dsandip** [comment]
@shahidhk is the right person to take this call. I am not sure if just an icon like [this](https://fontawesome.com/icons/edit?style=solid) will work and we may have to add text (`text area` ?)

**Comment by shahidhk** [comment]
What about something like this?

expand_ui

Icon used is https://fontawesome.com/icons/expand?style=solid

**Comment by Anupam-dagar** [comment]
@shahidhk does the placement of icon remains same for text area?

**Comment by shahidhk** [comment]
It has to be inside the textbox, top right corner.

File: console/src/components/Common/Common.scss [hunk 1]
```diff
@@ -20,6 +20,15 @@ table thead tr th
   color: #4D4D4D;
   font-weight: 600 !important;
 }
+
+.textareaButtonAlign {
+  position: relative;
+  float: right;
+  margin-left: -30px;
+  margin-right: 5px;
+  margin-top: 3px;
```
Comment (praveenweb) [inline console/src/components/Common/Common.scss]: margin-top: 10px;

File: console/src/components/Common/Toggler/Toggler.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            <input
+              {...this.props.standardProps}
+              placeholder={this.props.placeholderProp}
+            />
+          )}
+        </label>
+        <i
+          className={
+            this.state.input_textarea === 1
+              ? 'fa fa-compress fa-2x' + ' ' + styles.textareaButtonAlign
```
Comment (praveenweb) [inline console/src/components/Common/Toggler/Toggler.js]: let's remove fa-2x in these conditions

File: console/src/components/Common/Toggler/Toggler.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+              : 'fa fa-expand fa-2x' + ' ' + styles.textareaButtonAlign
```
Comment (praveenweb) [inline console/src/components/Common/Toggler/Toggler.js]: Remove fa-2x

**Comment by Anupam-dagar** [comment]
@shahidhk submitting a fix for the same.

**Comment by mnlbox** [comment]
@shahidhk any plan to merge this?

**Comment by Anupam-dagar** [comment]
@shahidhk I pass the tests on this in sometime.

**Comment by AnnieAtHasura** [comment]
> What about something like this?
> 
> expand_ui
> 
> Icon used is https://fontawesome.com/icons/expand?style=solid
> 
> We can also use https://fontawesome.com/icons/expand-alt?style=solid and https://fontawesome.com/icons/compress-alt?style=solid to indicate two states.

That icon's not obvious to me. I'll get design from Anu when I work on this.

**Comment by rikinsk** [comment]
Closing in favour of  #2498

**Pull request closed without merging**